desktop-widgets/tab-widgets/TabDiveInformation.cpp

```diff
@@ -20,6 +24,35 @@ TabDiveInformation::TabDiveInformation(QWidget *parent) : TabBase(parent), ui(ne
 	QStringList atmPressTypes { "mbar", get_depth_unit() ,"use dc"};
 	ui->atmPressType->insertItems(0, atmPressTypes);
 	pressTypeIndex = 0;
+	// This needs to be the same order as enum dive_comp_type in dive.h!
+	QStringList types = QStringList();
+	for (int i = 0; i < NUM_DIVEMODE; i++)
+		types.append(gettextFromC::tr(divemode_text_ui[i]));
+	ui->diveType->insertItems(0, types);
+	connect(ui->diveType, SIGNAL(currentIndexChanged(int)), this, SLOT(diveMode_Changed(int)));
```

Comment:
Please use modern pointer-to-function based syntax.
```
connect(ui->diveType, &QComboBox::currentIndexChanged, this, &TabDiveInformation::diveMode_Changed);
```

diveMode_Changed is a weird mix of camelCase and snake_case.

Reply:
Nevermind - this won't work owing to Qt's overloading of the currentIndexChanged signal. Just keep the old version.